src/component.ts

```diff
@@ -1,155 +1,175 @@
 export interface Component {
   id: string;
-  /// Returns true if the views associated with this component should be updated based on the state update.
-  needsRender(stateUpdate: StateUpdate): boolean;
-  /// Customize the HTMLElement & Style associated with this Component
-  /// Returns child components, if any
-  render(
-    store: Store,
-    element: HTMLElement,
-    style: Style,
-  ): Component[];
-}
 
-export class StateUpdate {
-  previousState: AppState;
-  currentState: AppState;
-  constructor(previousState: AppState, currentState: AppState) {
-    this.previousState = previousState;
-    this.currentState = currentState;
-  }
+  render(): void;
```

Comment:
@cyypherus - I didn't just work from an older version of the file.. I generally feel the original interface you made was better because I don't think the parent component needs to be in charge of special styling. I'd rather any special styling needs come through the controller, not the parent.